validate_email/smtp_check.py

```diff
@@ -169,35 +169,34 @@ def check(self, hosts: List[str]) -> bool:
             LOGGER.debug(msg=f'Trying {host} ...')
             if self._check_one(host=host):
                 return True
-        # Raise appropriate exceptions when necessary
-        if self.__communication_errors:
-            raise SMTPCommunicationError(
-                error_messages=self.__communication_errors)
-        elif self.__temporary_errors:
+        # Raise exception for collected temporary errors
+        if self.__temporary_errors:
             raise SMTPTemporaryError(error_messages=self.__temporary_errors)
 
 
 def smtp_check(
-        email_address: EmailAddress, mx_records: list, debug: bool,
-        from_address: Optional[EmailAddress] = None,
-        helo_host: Optional[str] = None, smtp_timeout: int = 10) -> bool:
+        email_address: EmailAddress, mx_records: List[str],
+        timeout: float = 10, helo_host: Optional[str] = None,
+        from_address: Optional[EmailAddress] = None, debug: bool = False
+        ) -> bool:
```

Comment:
This can look like
```python
def smtp_check(
    email_address: EmailAddress, mx_records: List[str],
    timeout: float = 10, helo_host: Optional[str] = None,
    from_address: Optional[EmailAddress] = None, debug: bool = False
) -> bool:
```

(autopep8 make these changes)